src/layouts/Dashboard/components/ReviewRequestModal/RequestOverview/RequestOverview.tsx

```diff
@@ -192,63 +192,131 @@ export const RequestOverview = ({
+      id: "type",
+      header: ({ column }) => (
+        <Menu>
+          <MenuButton
+            as={IconButton}
+            variant="clear"
             icon={<BiFilterAlt />}
-            variant="link"
+            fontSize="1.25rem"
+            aria-label="sort by file type"
           />
-        </Th>
+          <MenuList>
+            <MenuItem
+              icon={
+                // NOTE: Using an Icon component to hook into design system results in a
+                // slightly off center icon, which is why using the base component itself
+                // from react-icons + using the theme is preferred.
+                <BiFileBlank fontSize="1rem" fill={theme.colors.icon.alt} />
+              }
+              iconSpacing="0.5rem"
+              onClick={() => {
+                column.setFilterValue(["page", "image", "file"])
+              }}
+            >
+              <Text textStyle="subhead-2" textColor="text.body">
+                Pages
+              </Text>
+            </MenuItem>
+            <MenuItem
+              iconSpacing="0.5rem"
+              icon={<BiCog fontSize="1rem" fill={theme.colors.icon.alt} />}
+              onClick={() => {
+                column.setFilterValue(["setting"])
+              }}
+            >
+              <Text textStyle="subhead-2" textColor="text.body">
+                Settings
+              </Text>
+            </MenuItem>
+            <MenuItem
+              icon={<BiCompass fontSize="1rem" fill={theme.colors.icon.alt} />}
+              iconSpacing="0.5rem"
+              onClick={() => {
+                column.setFilterValue(["nav"])
+              }}
+            >
+              <Text textStyle="subhead-2" textColor="text.body">
+                Navigation
+              </Text>
+            </MenuItem>
+          </MenuList>
+        </Menu>
       ),
       cell: (props) => getIcon(props.row.original.type),
+      filterFn: "arrIncludesSome",
     }),
     columnHelper.accessor((row) => `${row.name} ${row.path}`, {
       id: "itemName",
+      sortingFn: "textCaseSensitive",
       cell: ({ row }) => (
         <ItemName name={row.original.name} path={row.original.path} />
       ),
-      header: () => (
-        <Th textTransform="capitalize">
-          <Text textStyle="subhead-2" textColor="text.label">
-            Item name
-          </Text>
-        </Th>
+      // NOTE: This is case insensitive and will search through both name + path
+      // This is due to the accessor returning a composite string.
+      // Do note that the path will be joined using a , so for composite search paths
+      // like: /some/folder, the correct search term would be some,folder.
+      filterFn: "includesString",
+      header: ({ column }) => (
+        <Button
+          ml="-0.25rem"
+          variant="link"
+          _hover={{
+            textDecoration: "none",
+          }}
+          onClick={column.getToggleSortingHandler()}
```

Comment:
Can I check for this sorting function, it can toggle between 3 states:
1. Unsorted
2. Sort ascending
3. Sort descending

Should we only allow it to toggle between states 2 and 3, and only state 1 happens when the user first loads the page?

Reply:
no strong opinion here, will defer to @NatMaeTan

Reply:
Is this for the last updated column? Or which column are you referring to? @dcshzj

Reply:
Both the item name and the last edited columns, should we allow the user to be able to go back to being unsorted?

Also, @NatMaeTan should we also change the icon to indicate the sorting order?

Reply:
^ wrt sorting order, nat say leave till later :/